src/AzureRenderManager/WebApp/Config/Coordinators/AssetRepoCoordinator.cs

```diff
@@ -93,5 +114,309 @@ public async Task<bool> RemoveRepository(AssetRepository repository)
+
+                    var templateParams = GetTemplateParameters(repository);
+
+                    var properties = new Deployment
+                    {
+                        Properties = new DeploymentProperties
+                        {
+                            Template = await _templateProvider.GetTemplate("linux-file-server.json"),
+                            Parameters = _templateProvider.GetParameters(templateParams),
+                            Mode = DeploymentMode.Incremental
+                        }
+                    };
+
+                    // Start the ARM deployment
+                    await client.Deployments.BeginCreateOrUpdateAsync(
+                        repository.ResourceGroupName,
+                        repository.DeploymentName,
+                        properties);
+
+                    // Queue a request for the background host to monitor the deployment
+                    // and update the state and IP address when it's done.
+                    await _deploymentQueue.Add(new ActiveDeployment
+                    {
+                        FileServerName = repository.Name,
+                        StartTime = DateTime.UtcNow,
+                    });
+
+                    repository.ProvisioningState = ProvisioningState.Running;
+                    repository.InProgress = false;
+
+                    await UpdateRepository(repository);
+                }
+            }
+            catch (CloudException ex)
+            {
+                Console.WriteLine($"Failed to deploy NFS server: {ex.Message}.\n{ex.StackTrace}");
+                throw;
+            }
+        }
+
+        private Dictionary<string, object> GetTemplateParameters(NfsFileServer repository)
+        {
+            var fileShare = repository.FileShares.FirstOrDefault();
+            return new Dictionary<string, object>
+            {
+                {"environmentTag", repository.EnvironmentName ?? "Global"},
+                {"vmName", repository.VmName},
+                {"adminUserName", repository.Username},
+                {"adminPassword", repository.Password},
+                {"vmSize", repository.VmSize},
+                {"subnetResourceId", repository.Subnet.ResourceId},
+                {"sharesToExport", fileShare?.Name ?? ""},
+                {"subnetAddressPrefix", string.Join(",", repository.AllowedNetworks)},
+            };
+        }
+
+        private static async Task IgnoreNotFound(Func<Task> action)
+        {
+            try
+            {
+                await action();
+            }
+            catch (CloudException e)
+            {
+                if (!ResourceNotFound(e))
+                {
+                    throw;
+                }
+            }
+        }
+
+        private static bool ResourceNotFound(CloudException ce)
+        {
+            return ce.Body.Code == "ResourceNotFound" || ce.Body.Code == "ResourceGroupNotFound";
```

Comment:
I think it's more robust to check HTTP status code first, we've had cases where the body didn't get populated properly in the past (e.g. from Storage APIs).